### ISSUE TITLE
Can not record alphanumeric keys in macOSX10

### ISSUE THREAD

**Issue by Freakwill** [moses-palmer/pynput#126]
Can not record alphanumeric keys in macOSX10. It had been discussed somewhere in the issues. But My issue is that it just did not print anything when I press keys except control/option/command... that lies on the boundary of the keyboard.

**Comment by oneTaken** [comment]
@Freakwill could you paste the complete code here or somewhere?
Your description sounds not easy to find your problem.

**Comment by Freakwill** [comment]
one of them is
```python
from pynput import keyboard

def on_press(key):
    try:
        print('alphanumeric key <{0}> pressed'.format(key.char))
    except AttributeError:
        print('special key <{0}> pressed'.format(key))

def on_release(key):
    print('{0} released'.format(key))
    if key == keyboard.Key.esc:
        return False

while True:
    with keyboard.Listener( on_press = on_press, on_release = on_release, suppress=True) as listener:
        listener.join()

```

**Comment by Freakwill** [comment]
@georgejdanforth NO! only the author can solve it. We have to wait new releases.

**Comment by moses-palmer** [comment]
Thank you for your reports.

I apologise for this very late response. I think that I may have a solution for you. There are actually two issues:

1. As described [in the documentation](https://pynput.readthedocs.io/en/latest/limitations.html#mac-osx), keyboard listeners are restricted on *OSX*. If your application is not whitelisted as detailed in in the link, you must run it as root, otherwise you will only get notifications about control keys. This is a very reasonable limitation, as you would otherwise be able to listen to password inputs.

2. Key presses with modifier keys are incorrectly handled in *pynput* on *OSX*. I pushed a test branch [here](https://github.com/moses-palmer/pynput/tree/fixup-osx-keyboard-listener) with an attempt at a fix.

I would appreciate if you could test the feature branch and see if it helps.
